daf/Configurator.cpp

```diff
@@ -33,29 +33,29 @@
 
 namespace DAF
 {
+    // NOTE: We can only use ACE::debug here as DAF::debug() may not have been set up yet and caused infinate recursion.
+
     namespace { // annanomous
 
         typedef std::map< Configurator::section_type, Configurator::property_list_type >  fileloader_map_type;
 
         struct ConfiguratorFileLoader : fileloader_map_type {
-            ConfiguratorFileLoader(const std::string &filename);
+            ConfiguratorFileLoader(const std::string & filename);
         };
 
-        ConfiguratorFileLoader::ConfiguratorFileLoader(const std::string &filename)
+        ConfiguratorFileLoader::ConfiguratorFileLoader(const std::string & filename)
         {
-            // NOTE: We can only use ACE::debug here as DAF::debug() may not have been set up yet.
-
             struct ConfigFile : std::ifstream {
-                ConfigFile(const std::string &file_name) : std::ifstream(file_name.c_str()) {
+                ConfigFile(const std::string & file_name) : std::ifstream(file_name) {
```

Comment:
Constructing a std::ifstream with a std::string isn't available until C++11. It was previously taking file_name.c_str() in order to use a const char*.